src/engine/TPA.cc

```diff
@@ -46,6 +47,16 @@ VerificationResult TPAEngine::solve(ChcDirectedHyperGraph const & graph) {
     auto translator = std::move(transformationResult.second);
     if (transformedGraph->isNormalGraph()) {
         auto normalGraph = transformedGraph->toNormalGraph();
+        if (options.hasOption(Options::SIMPLIFY_NESTED)) {
```

Comment:
Can you move this into the `solve` method that takes `DirectedChcGraph`?

Reply:
Updated